prov/rxm/src/rxm.h

```diff
@@ -672,9 +686,10 @@ rxm_process_recv_entry(struct rxm_recv_queue *recv_queue,
 static inline struct rxm_conn *
 rxm_acquire_conn(struct rxm_ep *rxm_ep, fi_addr_t fi_addr)
 {
-	return container_of(ofi_cmap_acquire_handle(rxm_ep->util_ep.cmap,
-						    fi_addr),
-			    struct rxm_conn, handle);
+	struct util_cmap_handle *handle =
+		ofi_cmap_acquire_handle(rxm_ep->util_ep.cmap, fi_addr);
+
+	return handle ? container_of(handle, struct rxm_conn, handle) : NULL;
```

Comment:
AFAIK, the handle (`ep->cmap->handles_av[fi_addr]`) is NULL only in case of `fi_av_remove` (expected for user) was called or remote shutdown (unexpected for user) was received.
This function is only called inside `rxm_acquire_conn_connect` function.
`rxm_acquire_conn_connect` has check for `rxm_conn` object for NULL. Adding this check we will have two check for the same case.
Can we move `handle` filed to the top of `rxm_conn` structure (+ add the comment that this should always be at the top). This give us possibility to avoid calling `container_of` and just cast `util_cmap_handle` to `rxm_conn`.
Caller of this function (`rxm_acquire_conn_connect`) must care to have a check for NULL.
```
static inline struct rxm_conn *
rxm_acquire_conn(struct rxm_ep *rxm_ep, fi_addr_t fi_addr)
{
        return (struct rxm_conn *)ofi_cmap_acquire_handle(rxm_ep->util_ep.cmap, fi_addr);
}
```
It should also save some instructions that is for the arithmetic operation applied for pointers. What do you think?

Reply:
> Can we move handle filed to the top of rxm_conn structure 

Yes, this seems like a good idea. I'll do it.